tools/wptrunner/wptrunner/executors/base.py

```diff
@@ -782,13 +782,14 @@ def process_action(self, url, payload):
         except self.unimplemented_exc:
             self.logger.warning("Action %s not implemented" % action)
             self._send_message(cmd_id, "complete", "error", f"Action {action} not implemented")
-        except self.expected_exc:
-            self.logger.debug(f"Action {action} failed with an expected exception")
-            self._send_message(cmd_id, "complete", "error", f"Action {action} failed")
-        except Exception:
-            self.logger.warning(f"Action {action} failed")
-            self._send_message(cmd_id, "complete", "error")
-            raise
+        except self.expected_exc as e:
+            exception_string = traceback.format_exc()
+            self.logger.debug(f"Action {action} failed with an expected exception:\n{exception_string}")
```

Comment:
optional nit: Would [`logger.debug(exc_info=True)` work][0] for delegating traceback formatting to the logger's formatter?

[0]: https://github.com/web-platform-tests/wpt/blob/eaf476812603bca784478cb874388fac6e265a18/tools/third_party_modified/mozlog/mozlog/structuredlog.py#L689

Reply:
Huh, I didn't know that existed (and we have a lot of usage already of `traceback.format_exc`), despite it being @jgraham who added it back in 2014: https://hg.mozilla.org/mozilla-central/rev/23b4eadbcc79d790980738cbdd49b817ca32e856.

Let me change to that while I fix the flake8 failures.